### ISSUE TITLE
Unable to upload saved game 

### ISSUE THREAD

**Issue by alicework** [Deevian/pathfinder-kingmaker-respec#18]
Hi, just tried the tool but seem like i can't upload the .zks file on either chrome or firefox :

On Chrome : Uncaught (in promise) TypeError: Cannot read property 'async' of null
    at loadJSONFilesFromSave.js:16

On Firefox : TypeError: t.file(...) is null -  loadJSONFilesFromSave.js:16:12
AS soon as i choose the file (or drag n drop it)

header.json is in the unzipped file

On OS X 10.14 (latest)

**Comment by alicework** [comment]
[Manual_40_Terres_vol_es___12__rastus__VII__4710___18_56_18.zip](https://github.com/Deevian/pathfinder-kingmaker-respec/files/2462573/Manual_40_Terres_vol_es___12__rastus__VII__4710___18_56_18.zip)
(changed extension to .zip for upload in github)

**Comment by Deevian** [comment]
Is it a quick save? The file needs to have a `header.json` and a `party.json`, otherwise this won't work.

**Comment by alicework** [comment]
No, a manual one. Well i tried with both :p

**Comment by Deevian** [comment]
Right, the issue is that I can't parse the JSON's in your save file. The fact that the file is a `zip` makes me think you might have messed around with it already?

**Comment by alicework** [comment]
Hi, i changed the extension of the file to upload it to github wich only supports specifics extensions when uploading. That save was unmessed with.

Here is a fresh save zipped. I tested on PC too (win10 and Kingmaker 1.0.7) and i've got the same errors on FF and Chrome. The saves work on either win or os x when i launch the game.

COuld it be that they changed how saves work on 1.0.7 ? Or is it just me ?

[Manual_44_LVL6.zks.zip](https://github.com/Deevian/pathfinder-kingmaker-respec/files/2468095/Manual_44_LVL6.zks.zip)

Anyway thanks for your time on this !

**Comment by Deevian** [comment]
Have you checked the Nexus mod? I'm planning on stopping developing this, and the mod seems to be functioning better and has more frequent updates.

**Comment by bepnye** [comment]
I had this same problem on OS X as well. It seems that the string returned by the reader isn't valid JSON - there was an extra initial character that looked like whitespace when I output it to console. Not sure why this is, but it was easily fixed by updating the file reader in `loadJSONFilesFromSave.js` to return `[r, JSON.parse(headerData.substring(1)), JSON.parse(partyData.substring(1))]`

**Comment by jarrard** [comment]
I got this problem under Linux, unfortunately there is no respec mod for Linux and also running via WINE DXVK does not work anymore for this game so native Linux version is all we got. 

Nexus mod site is loaded with strict windows only mods for the most part, not helpful.

Also is this the change you have in the JS file, gives me a syntax error with and without change.

```
import JSZip from "jszip/dist/jszip.min";
import { memoize } from "lodash";

/**
 * Loads the passed save file and handles the unzipping of the child JSON files.
 * This is also memoized to avoid re-loading the same save file multiple times.
 *
 * @param {Blob} file
 */
export default memoize((file) => {
    const reader = new JSZip();

    return reader.loadAsync(file)
        .then(() => Promise.all([
            reader,
            reader.file("header.json").async("string"),
            reader.file("party.json").async("string"),
        ]))
        .then(([r, headerData, partyData]) => [r, JSON.parse(headerData.substring(1)), JSON.parse(partyData.substring(1))]);
}, (file) => `${file.name}_${file.lastModified}`);

```

**Comment by Deevian** [comment]
@jarrard The tool has been deprecated, so I can't promise a fix for the issue, but -- I mildly recall that a recent change to the game has modified the save structure to reduce the file size (which was humongous, so it's understandable). Since the tool works under the assumption of a given file structure, it's very possible that this change might have broke it entirely.

If you want to look for a fix, I recommend going through the new save file structure to figure out where stuff is now being stored, and going forward from there. Do note that the tool still doesn't preserve some key character story changes (i.e. skills, feats, sprite change, etc), so use it at your own risk!

**Comment by jarrard** [comment]
Fair enough, I might hold out for possible respec option for the character editor which does compile and run on Linux (and other OS's) just fine. 

A little annoyed nexus modders have adopted a windows platform only approach for a multi-platform game.... infuriating really!

**Comment by Deevian** [comment]
I think it's just for that one guy (newman55), since the `Unity Mod Manager` is also his. Also, a bit old, but there seems to be a guy that was (or said he was) working on a fix for the mod manager in Linux in the Nexus bug page -- https://www.nexusmods.com/site/mods/21/?tab=bugs

Doesn't look like he actually did the PR, but the underlying problem does seem simple enough if you want to pick up from there.

**Comment by jarrard** [comment]
Cheers, might poke around see if I can get interest flowing again. I'm not all that useful when it comes to fixing these sort of problems, sometimes I get real lucky but so you never know.
